drain3/template_miner.py

```diff
@@ -170,24 +173,94 @@ def match(self, log_message: str, full_search_strategy="never") -> LogCluster:
         matched_cluster = self.drain.match(masked_content, full_search_strategy)
         return matched_cluster
 
-    def get_parameter_list(self, log_template: str, content: str):
-        escaped_prefix = re.escape(self.config.mask_prefix)
-        escaped_suffix = re.escape(self.config.mask_suffix)
-        template_regex = re.sub(escaped_prefix + r".+?" + escaped_suffix, self.drain.param_str, log_template)
-        if self.drain.param_str not in template_regex:
-            return []
-        template_regex = re.escape(template_regex)
-        template_regex = re.sub(r'\\ +', r'\\s+', template_regex)
-        template_regex = "^" + template_regex.replace(escaped_prefix + r"\*" + escaped_suffix, "(.*?)") + "$"
+    def get_parameter_list(self, log_template: str, log_message: str, exact_matching: bool = True) -> List[str]:
+        """
+        Extract parameters from a log message according to a provided template that was generated by Drain3 by calling `add_log_message()`.
+        This will work best if each `MaskingInstruction` has a `mask_with` unique across all instances
+        and none of them have a value of `*`.
+
+        :param log_template: log template corresponding to the log message
+        :param log_message: log message to extract parameters from
+        :param exact_matching: whether to apply the correct masking-patterns to match parameters, or try to approximate;
+            disabling exact_matching may be faster but may lead to situations in which parameters are wrongly identified
+        :return: An ordered list of parameter values present in the log message.
+        """
+        parameter_details_list = self.get_parameter_details_list(log_template, log_message, exact_matching=exact_matching)
+        return [parameter.value for parameter in parameter_details_list]
 
+    def get_parameter_details_list(self, log_template: str, log_message: str, exact_matching: bool = True) -> List[ParameterDetails]:
```

Comment:
I think it will be more elegant to use OrdereDict for the return value, or even ordinary dict which is ordered since Python 3.6+

Reply:
Also lets just call this function `ExtractParameters()` and make the `get_parameter_list()` deprecated just for backwards compatibility.

Reply:
taking back OrderedDict, we may have multiple params with same mask name

Reply:
If we make `get_parameter_list()` deprecated, I suggest keeping the old interface by removing the `exact_matching`-option and having it return `[]` when there is no match (as opposed to `None` for the new method).

Reply:
`ExtractedParameter` is the name of the type, right? How should the function be called, `extract_parameters`?

If we deprecate `get_parameter_list` I suggest keeping the old interface without the `exact_matching`-option and a return value of `[]` for no match (as opposed to `None` for the new method).
If `get_parameter_list` is truly deprecated, would also change the tests to only check the new method.

Reply:
Yes, agreed to all!